src/test/scala/mesosphere/marathon/state/VersionInfoTest.scala

```diff
@@ -96,6 +113,28 @@ class VersionInfoTest extends UnitTest {
       )
     }
 
+    "A restart change on FullVersion only changes lastConfigChangeAt" in {
+      Given("A FullVersionInfo")
+      val versionInfo = VersionInfo.FullVersionInfo(
+        version = Timestamp(1),
+        lastScalingAt = Timestamp(1),
+        lastConfigChangeAt = Timestamp(1)
+      )
+
+      When("Applying a restart change")
+      val newVersion = versionInfo.withRestartChange(Timestamp(2))
+
+      Then("lastConfigChangeAt is updated while lastScalingAt is not")
+      newVersion should be(
+        FullVersionInfo(
+          version = Timestamp(2),
+          lastScalingAt = Timestamp(1),
+          lastConfigChangeAt = Timestamp(2)
+        )
+      )
+      newVersion.lastConfigChangeVersion should equal(Timestamp(2))
```

Comment:
that's actually not needed once you realize that we only seem to use `VersionInfo` immutably. 

but better safe than sorry, i guess...